bindings/java/src/org/sleuthkit/datamodel/blackboardutils/CommunicationArtifactsHelper.java

```diff
@@ -0,0 +1,746 @@
+        
+        return addMessage(messageType, direction,
+                fromAddress,
+                Arrays.asList(toAddress),
+                dateTime, readStatus,
+                subject, messageText, threadId,
+                otherAttributesList);
+    }
+    
+     /**
+     * Adds a TSK_MESSAGE artifact. 
+     * 
+     * Also creates an account instance for the sender/receiver, and creates a 
+     * relationship between the self account and the sender/receiver account.
+     * 
+     * This method is for messages with a multiple recipients.
+     * 
+     * @param messageType message type, may be empty or null
+     * @param direction message direction, UNKNOWN if not available
+     * @param fromAddress sender address, may be null
+     * @param recipientsList recipient address list, may be null or empty list
+     * @param dateTime date/time of message, 0 if not available
+     * @param readStatus message read or not, UNKNOWN if not available
+     * @param subject message subject, may be empty or null
+     * @param messageText message body, may be empty or null
+     * @param threadId, message thread id, may be empty or null
+     * 
+     * 
+     * @return message artifact
+     */
+    public BlackboardArtifact addMessage( String messageType, 
+                CommunicationDirection direction, 
+                Account.Address fromAddress, 
+                List<Account.Address> recipientsList, 
+                long dateTime, MessageReadStatusEnum readStatus, 
+                String subject, String messageText, String threadId) {
+        return addMessage( messageType,  direction,  
+                          fromAddress,  recipientsList, 
+                          dateTime,  readStatus,  
+                          subject, messageText,  threadId, 
+                          Collections.<BlackboardAttribute>emptyList());
+    }
+    
+    /**
+     * Adds a TSK_MESSAGE artifact. 
+     * 
+     * Also creates an account instance for the sender/receiver, and creates a 
+     * relationship between the self account and the sender/receiver account.
+     * 
+     * This method is for messages with a multiple recipients.
+     * 
+     * @param messageType message type, may be empty or null
+     * @param direction message direction, UNKNOWN if not available
+     * @param fromAddress sender address, may be null
+     * @param recipientsList recipient address list, may be null or empty list
+     * @param dateTime date/time of message, 0 if not available
+     * @param readStatus message read or not, UNKNOWN if not available
+     * @param subject message subject, may be empty or null
+     * @param messageText message body, may be empty or null
+     * @param threadId, message thread id, may be empty or null
+	 * @param otherAttributesList other attributes, may be an empty list
+     * 
+     * 
+     * @return message artifact
+     */
+    public BlackboardArtifact addMessage( String messageType, 
```

Comment:
it was my understanding that the npath problems were to be addressed in this refactor. @rcordovano since this would not affect the public API, are we deferring this (or not doing it)?

Reply:
We are either deferring or not doing anything more; the latter is most likely at this time.